.github/workflows/release_and_publish.yml

```diff
@@ -0,0 +1,49 @@
+name: Release & Publish
+
+on:
+  push:
+    # Sequence of patterns matched against refs/tags
+    tags:
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Manual Release
+        default: test
+        required: false
+  
+
+jobs:
+  build:
+    name: Create release & publish to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      
+      - name: Set up python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.x"
+      
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine jupyter urllib3 pandas pyyaml
+          python setup.py sdist bdist_wheel
+      
+      - name: Create Release Notes
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            await github.request(`POST /repos/${{ github.repository }}/releases`, {
+              tag_name: "${{ github.ref }}",
+              generate_release_notes: true
+            });
+      
+      - name: Publish distribution 📦 to PyPI
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_PASSWORD }}
```

Comment:
I see that you have PyPI access, have you added a secret?

Reply:
Yes.